### PR TITLE
repository and other bug fixes

### PR DESCRIPTION
**Pull request by Rocco83**
Hi,

This is quite a huge set of changes.
- repository are managed now through one hash, as the upstream is changing data every time
- README improved for the repositories
- Gentoo support added (still to be fully tested)
- bugfix: FD Address must not be quoted. fixed in code and spec
- Debian 9 supported (and tested)
- Fedora several version supported (not tested) 

I've seen just now your approach for some of the changes i have made.
https://github.com/Project0/puppet-bareos/commit/3946765b87f08c1ad711602a39593b09b9043f1b

Please check mine and state for the best. I will fix my mull request eventually.
Maybe the best is the combination of the two.
Consider that $operatingsystemmajrelease is not always present, as facter has to be > 1.7.0 (or 1.8.0 someone claim, i have not checked the code)

### CODE REVIEW

**Comment by Project0** [comment]
Wow, thanks for the PR. 
I was unsure about the net-address stuff while implementing, the bareos documentation is sometimes unclear ;-).
I didnt put too much effort in the repository stuff since its still possible to manage the repo out of this module.

**Comment by Rocco83** [comment]
I have backported your commit. 
This was the best that is would sort out :-)

The repository is one of pain for bareos i think, and is also not maintained... 
but sometimes necessary.

Anyway... What's next?
Would you like to suggest some change in the code?
Waiting for some feedback!

File: manifests/repository.pp [hunk 1]
```diff
@@ -5,59 +5,124 @@
 # This class will be automatically included when a resource is defined.
 # It is not intended to be used directly by external resources like node definitions or other modules.
 class bareos::repository(
-  $release = 'latest'
+  $release = 'latest',
+  $repo_avail_hash = {},
+  $repo_manage_hash = undef,
```
[the remark this answers is not in the document]
Comment (Rocco83) [reply manifests/repository.pp]: I am using debian wheezy, jessie.
i want version at least 14, but best if 16.
jessie does, but version 14.
wheezy does not have any.
bareos repo contains packages for both debian, that will override the jessie one.

This array will let me manage bareos repo
- debian wheezy
- not debian jessie

technically, we can change the `$repo_avail_hash` directly, but this hash is meant for the devels. so i can upgrade the package and i will get the best out of the repo.

I was struggling with that for a long...

hope this clarify

**Comment by Rocco83** [comment]
@Project0 ping :)

File: manifests/repository.pp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Project0) [reply manifests/repository.pp]: sorry for the late response. I feel this is too much logic in this module. Why not using hiera for this kind of different configurations?

**Comment by Project0** [comment]
closing this (inactivity).

**Pull request closed without merging**